### PR TITLE
feat(compile): ability to embed local data files

### LINKED ISSUE #20885: Embed files when using deno compile
Is there a way to embed static files when compiling a deno project using `deno compile`?

I have not found any good solution to do this when I'm not in charge of accessing the filesystem. For context, I'd like to embed all of my SSR-templates and static files in the binary itself to make my web server easier to distribute. All of the solutions I have found so far (e.g. [leaf](https://deno.land/x/leaf@v1.0.4)) requires control over how files are accessed - which I don't have (nor want to have) when using a library such as Nunjucks.

### PR DESCRIPTION
**Pull request by dsherret**
This is now trivial to implement due to recent changes.

```
> deno compile --allow-read=. --include data-file.txt main.js
```

Slightly annoying to have to give read permissions. I feel like we should allow reading files in the binary by default.

This only applies to files on the filesystem. For remote modules, that's going to have to wait for `import ... from "./data.txt" with { "type": "bytes" }` or whatever it will be.

* Docs PR https://github.com/denoland/docs/pull/1164

Closes #20885

### CODE REVIEW

File: cli/standalone/binary.rs [hunk 1]
```diff
@@ -540,73 +553,79 @@ impl<'a> DenoCompileBinaryWriter<'a> {
...
       npm_vfs
     } else {
       VfsBuilder::new(root_path.clone())?
     };
+    for include_file in include_files {
+      let path = deno_path_util::url_to_file_path(include_file)?;
+      vfs
+        .add_file_at_path(&path)
+        .with_context(|| format!("Including {}", path.display()))?;
+    }
```
Comment (dsherret) [inline cli/standalone/binary.rs:626]: Files are included here.

File: cli/tools/compile.rs [hunk 2]
```diff
@@ -212,6 +214,48 @@ fn validate_output_path(output_path: &Path) -> Result<(), AnyError> {
   Ok(())
 }
 
+fn get_module_roots_and_include_files(
```
Comment (dsherret) [inline cli/tools/compile.rs:217]: Files are parsed out here. Only local files work.

**Comment by bartlomieju** [comment]
> Slightly annoying to have to give read permissions. I feel like we should allow reading files in the binary by default.

I agree about that. Any way we can fix it before landing? I guess we'd have to pass additional permissions when embedding metadata into the binary.

**Comment by dsherret** [comment]
Opened https://github.com/denoland/docs/pull/1164

**Pull request merged**